packages/plugins/typescript-resolvers/tests/mapping.spec.ts

```diff
@@ -0,0 +1,802 @@
+              name: 'Foo',
+              chats: [0,1,2],
+            };
+          }
+        },
+        Chat: {
+          id(parent) {
+            const id: number = parent;
+            return id;
+          }
+        }
+      }
+    `;
+
+    await validate(
+      [result, usage].join('\n\n'),
+      {
+        scalars: {
+          ID: 'number',
+        },
+      },
+      testSchema
+    );
+  });
+
+  it('Should build ResolversTypes with defaultMapper set using {T}', async () => {
+    const result = await plugin(
+      schema,
+      [],
+      {
+        noSchemaStitching: true,
+        defaultMapper: 'Partial<{T}>',
+      },
+      { outputFile: '' }
+    );
+
+    expect(result).toBeSimilarStringTo(`
+    export type ResolversTypes = {
+      Query: Partial<Query>,
+      MyType: Partial<MyType>,
+      String: Scalars['String'],
+      MyOtherType: Partial<MyOtherType>,
+      Subscription: Partial<Subscription>,
+      Boolean: Scalars['Boolean'],
+      Node: Partial<Node>,
+      ID: Scalars['ID'],
+      SomeNode: Partial<SomeNode>,
+      MyUnion: Partial<MyUnion>,
+      MyScalar: Scalars['MyScalar'],
+      Int: Scalars['Int'],
+    };`);
+  });
+
+  it('Should build ResolversTypes with defaultMapper set using {T} with external identifier', async () => {
+    const result = await plugin(
+      schema,
+      [],
+      {
+        noSchemaStitching: true,
+        defaultMapper: './my-wrapper#CustomPartial<{T}>',
+      },
+      { outputFile: '' }
+    );
+
+    expect(result).toBeSimilarStringTo(`import { CustomPartial } from './my-wrapper';`);
+    expect(result).toBeSimilarStringTo(`
+    export type ResolversTypes = {
+      Query: CustomPartial<Query>,
+      MyType: CustomPartial<MyType>,
+      String: Scalars['String'],
+      MyOtherType: CustomPartial<MyOtherType>,
+      Subscription: CustomPartial<Subscription>,
+      Boolean: Scalars['Boolean'],
+      Node: CustomPartial<Node>,
+      ID: Scalars['ID'],
+      SomeNode: CustomPartial<SomeNode>,
+      MyUnion: CustomPartial<MyUnion>,
+      MyScalar: Scalars['MyScalar'],
+      Int: Scalars['Int'],
+    };`);
+  });
+
+  it('Should map to a custom type on every level when {T} is used as default mapper', async () => {
+    const config = {
+      scalars: {
+        ID: 'number',
+      },
+      noSchemaStitching: true,
+      defaultMapper: 'Partial<{T}>',
+      mappers: {
+        User: 'number',
+      },
+    };
+    const testSchema = buildSchema(/* GraphQL */ `
+      type User {
+        id: ID!
+        name: String!
+        chats: [Chat!]
+      }
+
+      type Chat {
+        id: ID!
+        owner: User!
+        members: [User!]
+      }
+
+      type Query {
+        me: User
+      }
+    `);
+    const result = await plugin(testSchema, [], config, { outputFile: '' });
+
+    expect(result).toBeSimilarStringTo(`
+    export type ResolversTypes = {
+      Query: Omit<Partial<Query>, 'me'> & { me: Maybe<ResolversTypes['User']> },
```

Comment:
```suggestion
      Query: Partial<Omit<Query, 'me'> & { me: Maybe<ResolversTypes['User']>}>,
```

Shouldn't `Partial<T>` be on top of everything? Otherwise `me` is required.

Reply:
yeah you are right, i'll fix